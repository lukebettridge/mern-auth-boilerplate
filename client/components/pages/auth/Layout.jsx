import React from "react";
import PropType from "prop-types";

import { Container, Heading, Paragraph } from "../../styles";

const Layout = props => {
	return (
		<Container small>
			<Heading>Webapp</Heading>
			{props.children}
			<Paragraph center light mt="l">
				Developed by{" "}
				<a
					href="https://paddl.co.uk/"
					rel="noopener noreferrer"
					target="_blank"
				>
					Paddl
				</a>
				<br />. A limited company registered in England and Wales. Registered
				number: 11992253.
			</Paragraph>
		</Container>
	);
};

Layout.propTypes = {
	children: PropType.any
};

export default Layout;
