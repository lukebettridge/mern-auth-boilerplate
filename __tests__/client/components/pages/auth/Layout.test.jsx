import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Layout from "components/pages/auth/Layout";

describe("Auth Layout", () => {
	it("snapshot renders", () => {
		const subject = shallow(<Layout />);

		expect(toJSON(subject)).toMatchSnapshot();
	});

	it("renders children", () => {
		const subject = shallow(
			<Layout>
				<p>foo</p>
			</Layout>
		);

		expect(subject.find("p").length).toEqual(1);
		expect(subject.find("p").text()).toEqual("foo");
	});
});
