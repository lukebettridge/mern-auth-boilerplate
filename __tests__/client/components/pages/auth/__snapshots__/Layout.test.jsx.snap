// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Layout snapshot renders 1`] = `
<styles__Container
  small={true}
>
  <styles__Heading>
    Webapp
  </styles__Heading>
  <styles__Paragraph
    center={true}
    light={true}
  >
    Developed by
     
    <a
      href="https://paddl.co.uk/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Paddl
    </a>
    <br />
    . A limited company registered in England and Wales. Registered number: 11992253.
  </styles__Paragraph>
</styles__Container>
`;
